### DISCUSSION TITLE
[Feature] Tidy Up Sharing

### DISCUSSION THREAD

**Discussion by fritut08** [immich-app/immich#6335] [Feature Request]
### The feature

I am very happy that I learned of the existence of Immich. From what I have seen so far, it makes organizing photos and videos much easier than any other selfhosted alternative.

The partner sharing is a nice way of seeing the photos that my significant other has taken. But if I understand correctly, that is all that this feature lets me do. I can either view them under "Sharing" or have them integrated into my timeline (which can be activated in the settings of the web app or through an icon in the mobile app).

However, neither does it let me recognize faces in those pictures nor can I add those pictures to my (unshared) albums. To circumvent the second problem, I can create a shared album where each of us adds their respective pictures. In my account, this album is shown under "Albums" and under "Sharing" while the other user can only find this album under "Sharing". This seems a but clunky.

As users have pointed out in other forums ([Link 1](https://www.reddit.com/r/immich/comments/17rs415/partner_share_doesnt_share_everything/?rdt=58647), [Link 2](https://lemmy.elest.io/post/7451)), it would be nice to have a more powerful sharing mechanism which makes access to the partner's photos and albums easier.

If two accounts are mutual partners, I could imagine the following:

1. All photos of both accounts show up in the timeline by default. The photos uploaded by the other account have a little marker that indicates
2. In terms of FR and assignment to albums, it does not matter to which library the photos belong under the hood.
3. Shared albums show up in the "Album" section.

### Platform

- [X] Server
- [X] Web
- [X] Mobile

**Comment by fritut08** [comment]
Thinking about this, it occurs to me that this might become easier with nested albums as described [here](https://github.com/immich-app/immich/discussions/2073) or [here](https://github.com/immich-app/immich/discussions/5078). One could create a base album (e.g. "Our Photos") and share this with another account. Now, both accounts can add pictures from their own timeline to that album and then create sub-albums (e.g. "Ski trip 2023", "Vintage Cars", ...) using all the files within the parent album "Our Photos". This would even allow for some privacy since not the entire library is shared as with the current partner feature.

**Comment by bo0tzz** [comment]
#12614
